src/CMakeLists.txt

```diff
@@ -335,6 +335,11 @@ else()
   set(old_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
   set(BUILD_SHARED_LIBS FALSE)
   add_subdirectory(fmt)
+  if (WIN32)
+    # We'll use the header-only version for now in order to avoid
+    # duplicate symbol errors with llvm.
+    add_library(fmt::fmt ALIAS fmt-header-only)
```

Comment:
It's still needed, we're getting the following linker error otherwise:

```
ld.lld: error: fmt::v9::format_error::~format_error() was replaced
```

Reply:
Ah, it appears that these are different branches: `WITH_FMT_HEADER_ONLY` is acted on in `cmake/modules/Findfmt.cmake` only in the `fmt_FOUND` case while you are adjusting the `else` case here.  I wonder if instead of being conditioned on `WIN32` this should be made generic and be conditioned on `WITH_FMT_HEADER_ONLY`?

I have to confess that I don't understand the logic in `cmake/modules/Findfmt.cmake` though...

Reply:
@tchaikov It would be great if you acked this (a change to respect `WITH_FMT_HEADER_ONLY` when using the bundled version).

Reply:
it's marked resolved. so presumably all good?

Reply:
No, I marked as resolved to move this PR forward.  We would still very much appreciate your input here!

Reply:
@petrutlucian94 the prefix of the commit message is 

> common:

could you please use `cmake:`. it's not obvious to me that this change is changing the building system. 

> avoid duplicate symbols

also, i'd suggest extract this change into its own commit.  and the title should be more specific on what is changed. now it reads like the reversed version of bug report: a bug report is: compile fails with duplicated symbols

this extracted change addresses the inconsistency between the building with vendored fmtlib and that with system fmtlib package if `WITH_FMT_HEADER_ONLY` is defined, by defining an alias from `fmt::fmt` to `fmt-header-only`. but the funny thing is, `fmt/CMakeLists.txt` also defines a `fmt::fmt` alias library:
```cmake
add_library(fmt::fmt ALIAS fmt)
```

is it allowed to `add_library()` of the same name multiple times?

Reply:
Makes sense, I moved that to a separate commit and updated the commit message to use the ``cmake:`` prefix.

Cmake seems to allow redefining the alias and since it's the most convenient approach, I think we can stick with it.